From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dominik Csapak <d.csapak@proxmox.com>
Date: Mon, 20 Dec 2021 09:28:35 +0100
Subject: [PATCH] hide fullscreen button on 'isFullscreen' get variable

workaround for android app, since the webview does not allow
'requestFullscreen' apparently, so we make the webview fullscreen and
hide the button here.

Signed-off-by: Dominik Csapak <d.csapak@proxmox.com>
---
 app/pve.js | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/app/pve.js b/app/pve.js
index 4774e3a..583a406 100644
--- a/app/pve.js
+++ b/app/pve.js
@@ -12,6 +12,7 @@ export default function PVEUI(UI){
     this.nodename = WebUtil.getQueryVar('node');
     this.resize = WebUtil.getQueryVar('resize');
     this.cmd = WebUtil.getQueryVar('cmd');
+    this.fullscreen = WebUtil.getQueryVar('isFullscreen');
     this.lastFBWidth = undefined;
     this.lastFBHeight = undefined;
     this.sizeUpdateTimer = undefined;
@@ -313,6 +314,12 @@ PVEUI.prototype = {
 		.classList.add('pve_hidden');
 	}
 
+	// hide fullscren button when it's already fullscreen (e.g. android app)
+	if (me.fullscreen) {
+	    document.getElementById('noVNC_fullscreen_button')
+		.classList.add('pve_hidden');
+	}
+
 	// add command logic
 	var commandArray = [
 	    { cmd: 'start', kvm: 1, lxc: 1},
